### PR TITLE
Add isNaN function

### PR DESCRIPTION
**Pull request by JonathanMeans**

### CODE REVIEW

File: modules/c++/math/wscript [hunk 1]
```diff
@@ -4,7 +4,11 @@ VERSION         = '0.1'
 USELIB          = 'MATH'
 MODULE_DEPS     = 'sys'
 
-options = configure = distclean = lambda p: None
+options = distclean = lambda p: None
+
+def configure(conf):
+    conf.check_cc(function_name='isnan', header_name='math.h',
```
Comment (asylvest) [inline modules/c++/math/wscript]: Do this like we do in the sys wscript where it makes a math_config.h file.  The reasoning is that that way someone building their application without waf  (j.e. just linking in our stuff) will have the defines they need.

**Comment by JonathanMeans** [comment]
Good to go

File: modules/c++/math/include/math/Utilities.h [hunk 2]
```diff
@@ -51,6 +52,24 @@ inline double square(double val)
...
+/*!
+ * Return true if argument is NaN
+ *
+ * \param value Argument to be checked for NaN
+ * \return true if value is NaN
+ */
+template <typename T> bool isNaN(T value)
+{
+#ifdef HAVE_ISNAN
+    return isnan(value);
```
Comment (asylvest) [inline modules/c++/math/include/math/Utilities.h]: Include `math.h` to get the C `isnan()` function

File: modules/c++/math/wscript [hunk 3]
```diff
@@ -1,10 +1,19 @@
...
 VERSION         = '0.1'
 USELIB          = 'MATH'
 MODULE_DEPS     = 'sys'
 
-options = configure = distclean = lambda p: None
+options = distclean = lambda p: None
+
+
+def configure(conf):
+    def sys_callback(conf):
```
Comment (asylvest) [inline modules/c++/math/wscript]: Rename to `math_callback()`

**Comment by JonathanMeans** [comment]
Done.

File: modules/c++/math/include/math/Utilities.h [hunk 4]
```diff
@@ -51,6 +52,25 @@ inline double square(double val)
...
+/*!
+ * Return true if argument is NaN
+ *
+ * \param value Argument to be checked for NaN
+ * \return true if value is NaN
+ */
+template <typename T> bool isNaN(T value)
+{
+#ifdef HAVE_ISNAN
+    #include <math.h>
```
Comment (asylvest) [inline modules/c++/math/include/math/Utilities.h]: Put this up with the other includes.  I don't remember what happens if you include a header inside a namespace other than to say that there are times where it really really really doesn't do what you want, so you always want to put them up top.  For simplicity I wouldn't bother to put it inside a new ifdef up top; always including it is fine.

**Comment by JonathanMeans** [comment]
It's moved.

**Comment by asylvest** [comment]
Looks good!

**Pull request merged**